content/en/docs/setup/install/gateway/index.md

```diff
@@ -0,0 +1,228 @@
+---
+title: Installing Gateways
+description: Install and customize Istio Gateways.
+weight: 40
+keywords: [install,gateway,kubernetes]
+owner: istio/wg-environments-maintainers
+test: no
+---
+
+Along with creating a service mesh, Istio allows managing [gateways](/docs/concepts/traffic-management/#gateways),
+which run at the edge of the mesh and allow fine-grained control over traffic entering and leaving the mesh.
+
+Follow this guide to deploy one or more gateways.
+
+## Prerequisites
+
+This guide requires the Istio control plane [to be installed](/docs/setup/install/) before proceeding.
+
+## Installing the gateway with injection
+
+To support users that already utilize existing deployment tools, we provide a few different ways to deploy a gateway.
+Each method will result in the same outcome; we recommend choosing the method you are most familiar with.
+If you are not sure which to choose, we suggest the Simple YAML method.
+
+As a security best practice, it is recommended to deploy the gateway in different namespace from the control plane.
+
+Using the same mechanisms as [Istio sidecar injection](/docs/setup/additional-setup/sidecar-injection/#automatic-sidecar-injection), Istio allows the required
+options for gateways to run to be automatically injected.
+This enables developers to have full control over the gateway deployment, while simplifying operations - when a new upgrade is available, or a configuration has changed, gateway pods can simply be restarted to be updated.
+This makes the experience of operating a gateway deployment the same as operating sidecars.
+
+{{< tabset category-name="gateway-install-type" >}}
+
+{{< tab name="Simple YAML" category-value="yaml" >}}
+
+First, setup the Kubernetes configuration, called `ingress.yaml` here:
+
+{{< text yaml >}}
+apiVersion: v1
+kind: Service
+metadata:
+  name: istio-ingressgateway
+  namespace: istio-ingress
+spec:
+  type: LoadBalancer
+  selector:
+    istio: ingressgateway
+  ports:
+  - port: 80
+    name: http
+  - port: 443
+    name: https
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: istio-ingressgateway
+  namespace: istio-ingress
+spec:
+  selector:
+    matchLabels:
+      istio: ingressgateway
+  template:
+    metadata:
+      annotations:
+        # Enable gateway injection
+        inject.istio.io/templates: gateway
+      labels:
+        # Set a unique label for the gateway. This is required to ensure Gateway's
+        # can select this workload
+        istio: ingressgateway
+        # Enable gateway injection. If connecting to a revisioned control plane, replace with
+        # `istio.io/rev: <revision-name>`.
+        sidecar.istio.io/inject: "true"
```

Comment:
We need this label too?

Reply:
You can either label the entire namespace or label the pod itself. This is how injection works in general, and is not specific to gateway injection. I will make it more clear

Reply:
I added a note

Reply:
Yeah, using `sidecar.istio.io/inject` annotation for gateways feels wrong to me..

Reply:
I think the problem is that the `inject` annotation name is just too specific. It would be better named `proxy.istio.io/inject`. Could we add a new annotation (and maybe deprecate `sidecar.istio.io/inject`)?

Reply:
Aside from normal API migration pains, it would be extremely difficult to make this change. The mutating webhook already has 8 distinct label selection clauses in order to handle the complex (internal) logic we want to expose to users (in a hopefully simple to understand scheme) while keeping backwards compatibility. It would probably be exponentially worse to try to add another one.

It is a label btw, not an annotation

Reply:
I see your point. The name's not ideal, but it is what it is.